Makefile

```diff
@@ -76,6 +76,9 @@ clean:
 client: generate
 	hack/build-client.sh "${AUTOREST_IMAGE}" 2020-04-30 2021-09-01-preview 2022-04-01 2022-09-04 2023-04-01 2023-07-01-preview 2023-09-04 2023-11-22 2024-08-12-preview
 
+ci-rp: fix-macos-vendor
+	docker build . -f Dockerfile.ci-rp --ulimit=nofile=4096:4096 --build-arg REGISTRY=$(REGISTRY) --build-arg VERSION=$(VERSION) --no-cache=$(NO_CACHE)
```

Comment:
In the past, a couple of times we (non-ARO-SRE team) have been put into an inconvenient situation where we have to ensure podman also works while a lot of code/commands were already built that specifically used docker. While in theory podman can be a drop-in replacement of docker, the reality is that when docker is used, it's possible to unintentionally accumulate docker-specific features. In this case, when switching to podman, the amount of work necessary can be low to high, in my experience it was high or near high.

I'm not sure about the implications of this but I would highly recommend make the container engine a variable so both podman/docker usage are supporeted and tested.

Reply:
This is a good point. [Our current documentation for local development setup only includes instructions for podman + podman-docker](https://github.com/Azure/ARO-RP/blob/master/docs/prepare-your-dev-environment.md#software-required), so my guess is that nearly all devs are in a podman ecosystem and testing there first. At the same time, I imagine that our eventual CI pipeline runs will be using docker, because MSFT pipelines and gold-standard templates and build agents have been configured in that engine. Do you anticipate podman having features docker will not be able to support, or only the other way around?

Reply:
@SudoBrendan 

>Do you anticipate podman having features docker will not be able to support, or only the other way around? 

I cannot provide an exact answer for this. However at least in my experience, it's always the other way around. For example, the assumption of rootfull in docker that will not work in podman.

Reply:
I think that's mostly what I've seen as well - where at least podman has all functionality that docker has. If this changes in the future I think we can reassess - but agree this is something to keep in mind as a potential risk.